web/src/activity_ui.ts

```diff
@@ -13,30 +13,33 @@ import * as people from "./people";
 import * as pm_list from "./pm_list";
 import * as popovers from "./popovers";
 import * as presence from "./presence";
+import type {PresenceInfoFromEvent} from "./presence";
 import * as sidebar_ui from "./sidebar_ui";
 import {realm} from "./state_data";
 import * as ui_util from "./ui_util";
+import type {FullUnreadCountsData} from "./unread";
 import {UserSearch} from "./user_search";
 import * as util from "./util";
 
-export let user_cursor;
-export let user_filter;
+export let user_cursor: ListCursor<number> | undefined;
+export let user_filter: UserSearch | undefined;
 
 // Function initialized from `ui_init` to avoid importing narrow.js and causing circular imports.
-let narrow_by_email;
+let narrow_by_email: (email: string) => void;
 
-function get_pm_list_item(user_id) {
+function get_pm_list_item(user_id: string): JQuery | undefined {
     return buddy_list.find_li({
-        key: user_id,
+        key: Number.parseInt(user_id, 10),
     });
 }
 
-function set_pm_count(user_ids_string, count) {
+function set_pm_count(user_ids_string: string, count: number): void {
     const $pm_li = get_pm_list_item(user_ids_string);
+    assert($pm_li !== undefined);
```

Comment:
Hmm, I don't see why that assertion should always be true. We call this in a loop over every PM conversation that has unreads, and I guess I'm not sure why we would know that the buddy list contains all such items at all times (including the fact that this function gets called in the handler for when new things might be marked as unread).

Reply:
I wonder if this is causing bugs right now actually.

Because of [this change](https://github.com/zulip/zulip/commit/5db0f5f29af5825751ed8885aa1a2fd3e5de6969) we can now get `undefined`, but before got `[]`. And `update_unread_count_in_dom` calls `find` on the element right away.

I looked around and it seems like this is the only call to `buddy_list.find_li(` outside of the buddy list module, so I'll just fix it here.

I think what we want then is to only `ui_util.update_unread_count_in_dom` if the element is defined.

Reply:
Pushed with that

Reply:
Yeah I'd sorta have expected some Sentry output if this happened in practice; it might be that the current construction is such that it's a very unlikely race. This solution seems fine though.